girderformindlogger/api/v1/user.py

```diff
@@ -738,6 +739,29 @@ def updateUser(
 
         return self._model.save(user)
 
+    @access.user
+    @autoDescribeRoute(
+        Description("Update a user login.")
+        .param(
+            'login',
+            'Display name of the user, usually just their first name.',
+            default="",
+            required=False
+        )
+        .errorResponse()
+        .errorResponse(('You do not have write access for this user.',
+                        'Must be an admin to create an admin.'), 403)
+    )
+    def updateLogin(
+        self,
+        login=""
+    ):
+        user = self.getCurrentUser()
+        if len(login):
+            user['login'] = login
+
+        return self._model.save(user)
```

Comment:
We definitely don't want to expose the whole unfiltered User model, which includes the [userId](https://github.com/ChildMindInstitute/mindlogger-app-backend/issues/153), the [cryptographic salt](https://en.wikipedia.org/wiki/Salt_(cryptography)), and the user's cached applets.

I'm glad you put this in this PR however, because an existing deprecated route needs to be updated to prevent the exposure of the same sensitive data: https://github.com/ChildMindInstitute/mindlogger-app-backend/blob/76abb94e8214449e1bf19e6fd3718e603631b15f/girderformindlogger/api/v1/user.py#L739